DeepCat.js

```diff
@@ -424,11 +426,31 @@
 				.replace( /"\s*$/, '' )
 				.replace( /\\(?=.)/g, '' );
 		}
+		if( /~[0-9]+$/.test ( searchTerm ) ) {
+			var split      = searchTerm.split("~");
+			searchTerm = split[0];
+		}
 
 		return removeUnicodeNonPrintables( replaceWhiteSpace( searchTerm ) );
 	};
 
 	/**
+	 * @param {string} searchTerm
+	 * @return {string}
+	 */
+	DeepCat.extractDeepCatCategoryDepth = function( searchTerm ) {
+		let depth = defaultDepth;
+		searchTerm = searchTerm.trim();
+
+		if( /~[0-9]+$/.test ( searchTerm ) ) {
+			var split      = searchTerm.split("~");
```

Comment:
This splitting is done in two different places. It should be a single method that returns an array with two elements.

Here is a regex you can use:
```js
/^\s*(.*?)\s*(?:~\s*(\d+)\s*)?$/.exec( 'term~2' )
```